### DISCUSSION TITLE
Power BI Premium + TE3 + Oracle + Gateway

### DISCUSSION THREAD

**Discussion by mariussveen** [TabularEditor/TabularEditor3#391] [🙋♀️TE3 Q&A] [answered]
Hi,

I have been using TE3 for a year with Power BI Premium and Azure Synapse (Dedicated SQL Pool) and that has worked like a charm. I'm now in a new job, and here we are using Oracle onprem with Power BI Premium and a Gateway.

I want the team to adapt TE3, but have some questions about how TE3 works with Oracle and Gateway.

First thing i noticed is that the import table wizard is not working.
See screenshots:
![wizard1](https://user-images.githubusercontent.com/98900996/152191440-ae169431-36e0-483e-82b7-c4e6a0f58671.PNG)
![wizard2](https://user-images.githubusercontent.com/98900996/152191455-814f85d7-d591-4c5d-a7a8-4af75446d9e7.PNG)
![wizard3](https://user-images.githubusercontent.com/98900996/152191465-692a6303-6770-4cdf-9ae5-5edfe66db817.PNG)

This is how my m-code looks like:
![mcode](https://user-images.githubusercontent.com/98900996/152191540-0f10a22a-16eb-4799-87f9-a09abb3fc1f3.PNG)

This works like a charm in the service, so its only in TE3 its a issue.

I'm also having trouble when using the "Update table schema" function. I guess its something simular with the issue over.

BR
Marius

**Comment by otykier** [comment, marked as answer]
We do not yet support all types of oracle sources natively. Please take a look at the workaround described here: https://docs.tabulareditor.com/te3/tutorials/importing-tables.html#updating-table-schema-through-analysis-services

**Reply by mariussveen** [reply]
Thank you so much Daniel, it works just like expected.

Marius

**Reply by otykier** [reply]
Hi @mariussveen - just an update on this. We now (finally!) support Oracle natively. Take a look at the release notes of [3.5.0](https://docs.tabulareditor.com/te3/other/release-notes/3_5_0.html?tabs=downloads).

**Reply by mariussveen** [reply]
Hi @otykier,

Thanks for the heads up, I tried this today. Works really well, and the performance is significant better. This was a really Nice suprise!!

Marius
